### ISSUE TITLE
Support self signed certificate generation

### ISSUE THREAD

**Issue by mattnelson** [agileorbit-cookbooks/java-libraries#7]
When deploying services in a private(or testing) environment, it would be beneficial if we could generate a self signed certificate. Is this something you would be willing to accept a pull request for?

**Comment by erichelgeson** [comment]
Yes, sound great. Please try to limit any use of non-standard commands (so it can be used in most platforms). Be happy to review that PR.
